v2/cmd/nuclei/main.go

```diff
@@ -160,6 +160,7 @@ on extensive configurability, massive extensibility and ease of use.`)
 		flagSet.BoolVarP(&options.ShowMatchLine, "show-match-line", "sml", false, "show match lines for file templates, works with extractors only"),
 		flagSet.BoolVar(&options.ZTLS, "ztls", false, "use ztls library with autofallback to standard one for tls13"),
 		flagSet.StringVar(&options.SNI, "sni", "", "tls sni hostname to use (default: input domain name)"),
+		flagSet.StringVarP(&options.Interface, "interface", "i", "", "network interface to use for network scan"),
```

Comment:
I think the user should specify either:
- Source Ip (we need to ensure that at least one interface exists with such source IP)
- Interface and interface's Source IP (we need to ensure that the specified interface has the desired source IP)